css-pseudo-4/Overview.bs

```diff
@@ -883,12 +885,17 @@ Additions to the CSS Object Model</h2>
   The <dfn attribute for=CSSPseudoElement>element</dfn> attribute is the
   [=originating element=] of the pseudo-element.
 
+  The <dfn attribute for=CSSPseudoElement>style</dfn> attribute
+  is [=CSS declaration block=] whose associated properties
+  are as follows:
+  * the [=readonly flag=] is unset,
+  * the [=parent CSS rule=] is <code>null</code>,
+  * the [=owner node=] is the pseudo-element
+  * the initially empty [=declarations=] list represents a [=style attribute=] for this pseudo-element
+          in the sence of [[!css-style-attr]] and as used in [[!css-cascade-4]].
+
   Note: This interface may be extended in the future
-  to other pseudo-element types
-  and/or to allow setting style information
-  through a {{CSSStyleDeclaration}} <code>style</code> attribute.
-  The current functionality is limited
-  to that which is needed to support [[web-animations-1]].
+  to other pseudo-element types.
```

Comment:
The CSSPseudoElement interface currently only supports `::before`, `::after` and `::marker` (lines 876-879). These are all [tree-abiding pseudo elements](https://www.w3.org/TR/css-pseudo-4/#treelike), meaning that they have a well-defined tree relationship with their owning element, and as a result, we can craft a sensible event flow when one of these receive an event.

CSSPseudoElement does not support (at least yet!) any pseudo elements that inherit styles from other pseudo elements, like ::selection or ::highlight, and not even pseudos that apply styles to multiple elements, like ::first-line. These seem harder to support because it's not clear what the event flow should be and how we would implement it (related issue: #4398 ). If we keep the same event path as we do currently, we probably won't deliver events to non-owning DOM elements that are covered by the pseudo. We would need to define a new event path and make sure it also works for other types of events (like pointer events - these are important for [highlight events](https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/master/highlight/events-explainer.md)).

I'm happy to brainstorm about this further (assuming people want to expand CSSPseudoElement to non-tree abiding elements). If either you and @fantasai have any ideas or know of any discussions that have taken place about this before, I would love to hear them. I'll also ponder on this and try to bring something to discuss during our next highlight API discussion call.